indico/htdocs/js/indico/jquery/colorswitch.js

```diff
@@ -0,0 +1,93 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    $.widget('indico.colorswitch', {
+
+        options: {
+            defaultColors: [],
+            onSelect: null
+        },
+
+        _create: function() {
+            var self = this;
+            var element = this.element;
+            var palette = $('<div>', {'class': 'color-palette', 'css': {'display': 'none'}});
+            var paletteTable = $('<table></table>');
+            var trigger = this.element.find('.switch-trigger');
+            var defaultColors = this.options.defaultColors;
+            var tr = this._createTableRow();
+
+            for (var i = 0; i < defaultColors.length; ++i) {
+                var td = $('<td></td>', {
+                    'css': {'background': defaultColors[i][1]},
+                    'class': 'palette-color'
+                });
+
+                var colorBox = $('<div></div>', {
+                    'value': defaultColors[i][1],
+                    'class': 'background-box'
+                });
+
+                colorBox.append($('<div></div>', {
+                    'css': {'background': defaultColors[i][0]},
+                    'class': 'text-box',
+                    'value': defaultColors[i][0]
+                }));
+
+                td.on('click', function() {
+                    var $this = $(this),
+                        backgroundColor = $this.find('.background-box').val(),
+                        textColor = $this.find('.text-box').val();
+
+                    trigger.css('background', backgroundColor)
+                           .css('color', textColor + ' !important');
+
+                    if (self.options.onSelect) {
+                        self.options.onSelect.call(trigger, backgroundColor, textColor);
```

Comment:
does it make sense to use the trigger as the context and not `self` or `self.element`?